### PR TITLE
Implements map of electronics gifting cities

### PR DESCRIPTION
**Pull request by ozer619**
We need to put the google api key in place of the env variable ,I have implemented the map which marks cities which donated laptops , please guide further.

### CODE REVIEW

**Review by RedFox0x20 — CHANGES_REQUESTED** [review]
Looks good to me! Had a brief chance to see that the pins were showing up before it kicked me out for no API key.

The following isn't crucial before merging however ideally it would be done to prevent users who access the page having errors:
I believe either @audreyfeldroy or @pydanny will have to add the API keys due to the Google Accounts in use.

**Comment by audreyfeldroy** [comment]
Looking into getting an API key for `REACT_APP_GOOGLE_KEY`. I think that's [Maps JavaScript API](https://console.cloud.google.com/apis/library/maps-backend.googleapis.com)?

File: pages/electronics-gifting-cities.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  Marker,
+  withGoogleMap,
+  withScriptjs,
+} from 'react-google-maps';
+
+function Map() {
+  const [selectedcity, setSelectedcity] = useState(null);
+
+  return (
+    <GoogleMap defaultCenter={{ lat: 34.5794, lng: -118.1164 }} defaultZoom={4}>
```
Comment (marekrozmus) [inline pages/electronics-gifting-cities.js:17]: maybe we could keep it in some const with name of the place?

File: pages/electronics-gifting-cities.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+
+import * as laptopCitiesData from '../data/laptop-cities.json';
+
+import {
```
Comment (marekrozmus) [inline pages/electronics-gifting-cities.js:5]: Please keep external imports together in group

File: pages/electronics-gifting-cities.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  GoogleMap,
+  InfoWindow,
+  Marker,
+  withGoogleMap,
+  withScriptjs,
+} from 'react-google-maps';
+
+function Map() {
+  const [selectedcity, setSelectedcity] = useState(null);
```
Comment (marekrozmus) [inline pages/electronics-gifting-cities.js:14]: Please change to camel case: `selectedCity` and `setSelectedCity`

**Review by marekrozmus — CHANGES_REQUESTED** [review]

File: pages/electronics-gifting-cities.js (hunk 1 above)
Comment (audreyfeldroy) [reply pages/electronics-gifting-cities.js:17]: Good feedback @marekrozmus. 

In this project, we're aiming to have all code follow the [ESLint no-magic-numbers rule](https://eslint.org/docs/rules/no-magic-numbers). 

Try this @ozer619 to make your code more explicit:

1. Define a `const` with a name like `MAP_CENTER_MHF_OFFICE` per [ES6 naming conventions].(https://stackoverflow.com/questions/40291766/naming-convention-for-const-object-keys-in-es6/40291962)
2. Add a comment like `// The MHF office at 1543 E Palmdale Blvd. Suite E, Palmdale, CA 93550` to help other developers understand what the `lat`, `lng` numbers are referring to.

**Review by audreyfeldroy — CHANGES_REQUESTED** [review]
Great work @ozer619. Try adding commits to address @marekrozmus's feedback as well, and let us know if you need any help. You're doing well and we're all here for you.

In the meantime I'll keep trying to get my Maps API key to work locally...

**Review by RedFox0x20 — APPROVED** [review]
This looks great! Thanks for making the changes suggested by @marekrozmus

I updated `docs/env_vars.md` to contain the correct image paths and giving the images an appropriate size.

**Review by marekrozmus — CHANGES_REQUESTED** [review]

File: pages/electronics-gifting-cities.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+} from 'react-google-maps';
+import React, { useState } from 'react';
+
+import * as laptopCitiesData from '../data/laptop-cities.json';
+
+function Map() {
+  const [selectedCity, setSelectedCity] = useState(null);
+  ///The MHF office at 1543 E Palmdale Blvd. Suite E, Palmdale, CA 93550 to (Default zoom)
+  const MAP_CENTER_MHF_OFFICE = [34.5794, -118.1164];
+  const DEFAULT_ZOOM = 4;
```
[the remark this answers is not in the document]
Comment (RedFox0x20) [reply pages/electronics-gifting-cities.js:16]: Why would it be necessary to move them outside of the component if that's the only scope they will be used in?

[the remark this answers is not in the document]
Comment (marekrozmus) [reply pages/electronics-gifting-cities.js:16]: It is like a good practice with react components. On every render these consts are recreated.
The scope here is the file itself so they won't "leak" anywhere :)

But it is just a suggestion 🙂

[the remark this answers is not in the document]
Comment (RedFox0x20) [reply pages/electronics-gifting-cities.js:16]: Ah, okay, that makes a lot of sense! TIL.

**Review by audreyfeldroy — CHANGES_REQUESTED** [review]
Hey, how's it going @ozer619? I noticed the API key in the diff has the old name without the `NEXT_PUBLIC` prefix. Kindly review my suggestions and either accept or correct them as needed.

**Comment by ozer619** [comment]
> Hey, how's it going @ozer619? I noticed the API key in the diff has the old name without the `NEXT_PUBLIC` prefix. Kindly review my suggestions and either accept or correct them as needed.

I'm really sorry for the inconvenience i should have double checked it, work on it now.

File: docs/env_vars.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+`NEXT_PUBLIC_GMAPS_JS_API_KEY `
+
+API key for the Maps JavaScript API. 
+
+### Getting a Key
+
+Instructions for getting this key: [Maps JavaScript API: Using API Keys > Creating API Keys](https://developers.google.com/maps/documentation/javascript/get-api-key#creating-api-keys)
+
+In the Google console, name it something like **REACT_APP_GOOGLE_KEY (Dev, audreyfeldroy)** or **REACT_APP_GOOGLE_KEY (Prod)**:
```
Comment (audreyfeldroy) [inline docs/env_vars.md:22]:
```suggestion
In the Google console, name it something like ** NEXT_PUBLIC_GMAPS_JS_API_KEY (Dev, audreyfeldroy)** or ** NEXT_PUBLIC_GMAPS_JS_API_KEY (Prod)**:
```

**Review by audreyfeldroy — CHANGES_REQUESTED** [review]
Found 2 more instances of the old key name and suggested fixes 🙂

File: docs/env_vars.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+![Google Maps JS API Key Naming](./images/gmaps-key-naming.png)
+
+That will help you identify which key is for which environment.
+
+### Key Restrictions
+
+If you're just using this key for local development, it's okay to leave *Key restrictions* as None like this:
+
+<img alt="Google Maps JS API Key Restrictions: Dev Environment" src="./images/gmaps-key-restrictions-dev.png" width="600"/>
+
```
Comment (audreyfeldroy) [inline docs/env_vars.md:33]: For the Vercel preview key, the *Website restrictions* setting will be something like:

```
*-margaritahumanitarian.vercel.app/*
```

**Review by audreyfeldroy — APPROVED** [review]

**Pull request merged**